R/survey_weights.R

```diff
@@ -0,0 +1,227 @@
+#'           and finally at the household level you either interview one person
+#'           or everyone in that household.
+#'           So the population no longer matters,
+#'           instead of being population counts as in the above stratified example,
+#'           you have a list of all clusters in the area with the number of houses
+#'           within each cluster.
+#'           Your survey dataset is collected with two levels, one for household
+#'           and one for individual. But those two levels are merged, so for
+#'           each individual there are two variables included: number of people in
+#'           household, and number of people surveyed.
+#'           To calculate the weight for each individual, you multiply the inverses
+#'           (i.e. 1 over) the probability of: a cluster being chosen,
+#'           a household being chosen in that cluster, an individual being chosen
+#'           in their household.
+#'
+#'
+#' @param x a data frame of survey data
+#' @param p a data frame containing popuplation data for groups in `...`
+#' @param cz a data frame containing a list of clusters and the number of
+#' households in each. Default is NULL as the default `method` is "stratified".
+#' @param ... shared grouping columns across both `x` and `p`. These are used
+#'   to match the weights to the correct subset of the pouplation.
+#'   Used only when `method` is "stratified".
+#' @param population the column in `p` that defines the pouplation numbers
+#' @param cluster_cz the column in `cz` that lists all possible clusters.
+#' Ignored if `method` is not "cluster" or if `ignore_cluster`
+#' is TRUE.
+#' @param household_cz the column in `cz` that lists the number of households per cluster.
+#' Ignored if `method` is not "cluster" or if `ignore_household`
+#' is TRUE.
+#' @param cluster_x the column in `x` that indicates which cluster rows belong to.
+#' Ignored if `method` is not "cluster" or if `ignore_cluster`
+#' is TRUE.
+#' @param household_x the column in `x` that indicates a unique household identifier.
+#' Ignored if `method` is not "cluster or if `ignore_household` is TRUE.
+#' @param individuals_eligible_x the column in `x` which specifies the number of people
+#' eligible for being interviewed in that household. (e.g. the total number of children)
+#' @param individuals_interviewed_x the column in `x` which specifies the number of people
+#' actually interviewed in that household.
+#' @param method what type of survey method you would like to use. Options are
+#' "stratified" (default) and "cluster".
+#' Stratified: takes variables grouped population counts from `p` and divides
+#' the corresponding groups by counts of variables in `...` in `x`, to create a weight.
+#' Cluster: Will multiply the inverse chances of a cluster being selected, a household
+#' being selected within a cluster, and an individual being selected within a household.
+#' As follows:
+#' unique(cluster_cz) / unique(cluster_x) *
```

Comment:
When you are combining calculations, please put braces around the parts that need to be evaluated first. I'm assuming that this means:

```
(unique(cluster_cz) / unique(cluster_x)) * (unique(household_cz) / unique(household_x)) * (individuals_eligible_x / individuals_interviewed_x)
```